feat(service): add autobase template

vitabaks opened the pull request:
Adds a service template for Autobase, an open-source self-hosted DBaaS for PostgreSQL, allowing easy deployment via Coolify.

<img width="3584" height="2130" alt="image" src="https://github.com/user-attachments/assets/70d35252-d5fa-4887-9c15-caa8a048ecdf" />

<img width="3584" height="2130" alt="image" src="https://github.com/user-attachments/assets/9d182f38-0ec7-4fa8-b629-fb701f89579f" />

vitabaks committed:
feat: add Autobase service template

Adds a service template for Autobase, an open-source self-hosted DBaaS for PostgreSQL, allowing easy deployment via Coolify.

vitabaks commented:
I tested it, and everything works. The only issue is that I couldn’t get the `PG_CONSOLE_API_URL` variable to populate automatically.
So for now, the `API_URL` variable needs to be set manually before deploying the service.

<img width="2292" height="876" alt="image" src="https://github.com/user-attachments/assets/faf64fcd-cbe9-4a26-b826-67ba48ead7e5" />

Result:

<img width="3584" height="2130" alt="image" src="https://github.com/user-attachments/assets/f356fad4-b265-4dcc-811b-1ea03d963fc2" />
<img width="3584" height="2130" alt="image" src="https://github.com/user-attachments/assets/1b7bd223-eb39-4aba-bb33-888dcaef1c50" />

peaklabs-dev commented on templates/compose/autobase.yaml:
```diff
@@ -0,0 +1,57 @@
+      interval: 5s
+      timeout: 5s
+      retries: 10
+    restart: unless-stopped
```
remove all `restart` as coolify will add them

vitabaks replied:
removed

peaklabs-dev commented on templates/compose/autobase.yaml:
```diff
@@ -0,0 +1,57 @@
+
+services:
+  autobase-console-db:
+    image: autobase/console_db:2.3.0
```
if this is just postgres please add the postgres ENVs

vitabaks replied:
what ENVs should be added?

This is a custom postgres image https://github.com/vitabaks/autobase/blob/master/console/db/Dockerfile

vitabaks replied:
Added a variable that allows to change the password for the database.

peaklabs-dev commented on templates/compose/autobase.yaml:
```diff
@@ -0,0 +1,57 @@
+      retries: 10
+    restart: unless-stopped
+    volumes:
+      - ${COOLIFY_VOLUME_APP}:/var/lib/postgresql
```
Currently ENVs are not supported in volume names. Please use a named volume

vitabaks replied:
renamed to console_postgres

peaklabs-dev commented on templates/compose/autobase.yaml:
```diff
@@ -0,0 +1,57 @@
+    environment:
+      - SERVICE_FQDN_AUTOBASE_CONSOLE_API_8080
+      - PG_CONSOLE_DB_HOST=autobase-console-db
+      - PG_CONSOLE_AUTHORIZATION_TOKEN=${AUTH_TOKEN:-secret_token}
```
Can this token be autogenerated?

vitabaks replied:
Ideally, Coolify would auto-generate a token while letting users redefine it. I’m not sure about the implementation details yet.

vitabaks replied:
@peaklabs-dev Can coolify generate a token and substitute it as the value of the AUTH_TOKEN variable? Something like: `${AUTH_TOKEN:-SERVICE_BASE64_AUTOBASE_CONSOLE_UI}`

The user must have access to the value of this variable for viewing, so that they can copy it and log in to the UI. Otherwise, we can leave the default value `secret_token`.

vitabaks replied:
figured it out

`SERVICE_PASSWORD_UI`

peaklabs-dev commented on templates/compose/autobase.yaml:
```diff
@@ -0,0 +1,57 @@
+      - PG_CONSOLE_DB_HOST=autobase-console-db
+      - PG_CONSOLE_AUTHORIZATION_TOKEN=${AUTH_TOKEN:-secret_token}
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
```
Why does autobase need access to the docker socket? This is a big security risk.

vitabaks replied:
Since the Autobase Console launches a container for deployment, you can read more about it here: https://github.com/vitabaks/autobase/blob/master/console/README.md

peaklabs-dev commented on templates/compose/autobase.yaml:
```diff
@@ -0,0 +1,57 @@
+    environment:
+      - SERVICE_FQDN_AUTOBASE_CONSOLE_UI_80
+      - PG_CONSOLE_AUTHORIZATION_TOKEN=${AUTH_TOKEN:-secret_token}
+      - PG_CONSOLE_API_URL=${API_URL:-SERVICE_FQDN_AUTOBASE_CONSOLE_API/api/v1}
```
Just use `SERVICE_FQDN_AUTOBASE_CONSOLE_API` then it will populate the API url

vitabaks replied:
I believe I tried this, but it didn’t work. I’ll make more time for further testing.

vitabaks replied:
I have released a new [release](https://github.com/vitabaks/autobase/releases/tag/2.3.2), now there is no need to publish the API.

vitabaks committed:
Remove restart

vitabaks committed:
rename db volume

vitabaks committed:
Update autobase to v2.3.2

vitabaks committed:
add environment for autobase-console-db

vitabaks committed:
more envs for autobase-console-api

vitabaks committed:
Add links to README

vitabaks committed:
formatting

vitabaks committed:
Use SERVICE_PASSWORD_AUTOBASE_CONSOLE_DB

vitabaks committed:
SERVICE_BASE64 for ENCRYPTIONKEY

vitabaks committed:
update service vars

vitabaks committed:
Generate AUTHORIZATION_TOKEN

vitabaks committed:
SERVICE_BASE64_ENCRYPTIONKEY

vitabaks commented:
Please review and merge.

vitabaks commented:
@peaklabs-dev any updates? 🙂

vitabaks committed:
Merge branch 'next' into v4.x

github-actions[bot] committed:
docs: update changelog

vitabaks committed:
Update autobase to v2.4.1

github-actions[bot] committed:
docs: update changelog

vitabaks commented:
Updated to version [2.4](https://github.com/vitabaks/autobase/releases/tag/2.4.0)

vitabaks committed:
Merge branch 'next' into v4.x

github-actions[bot] committed:
docs: update changelog

vitabaks committed:
Update CHANGELOG.md

github-actions[bot] committed:
docs: update changelog

vitabaks committed:
chore: exclude changelog from PR

github-actions[bot] committed:
docs: update changelog

peaklabs-dev committed:
Merge branch 'next' into v4.x

github-actions[bot] committed:
docs: update changelog

peaklabs-dev committed:
refactor(service): improve autobase

- fix formatting, naming and order
- fix add platform: linux/amd64 to avoid issues

github-actions[bot] committed:
docs: update changelog

peaklabs-dev reviewed (approved):
Thanks for the PR.

It would be cool if you could upgrade it to v2.5 via another PR.

And next time please do not open a PR from `v4.x` or `next` from your repo.

vitabaks commented:
> It would be cool if you could upgrade it to v2.5 via another PR.

I'll do it, version 2.5 is really good.

> And next time please do not open a PR from v4.x or next from your repo.

Ok. Thanks!
